src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -703,11 +703,14 @@ public static IEnumerable<TSource> SkipWhile<TSource>(this IEnumerable<TSource>
 
         private static IEnumerable<TSource> SkipWhileIterator<TSource>(IEnumerable<TSource> source, Func<TSource, bool> predicate)
         {
-            bool yielding = false;
-            foreach (TSource element in source)
+            using (IEnumerator<TSource> e = source.GetEnumerator())
             {
-                if (!yielding && !predicate(element)) yielding = true;
-                if (yielding) yield return element;
+                do
+                    if (!e.MoveNext()) yield break;
+                while (predicate(e.Current));
+                do
+                    yield return e.Current;
+                while (e.MoveNext());
```

Comment:
@stephentoub Yep, [here they are](https://gist.github.com/jamesqo/87e3ab99180f4f01cc30) ([results](https://gist.github.com/jamesqo/05aed6873cbd6812b406)). Unfortunately, it looks like nguerrera/Maxwe11 are correct; the new code is barely faster than the old code.